### DISCUSSION TITLE
is it possible to change dtype to uint16?

### DISCUSSION THREAD

**Discussion by oswald1234** [corteva/geocube#98] [Q&A] [answered]
I'm trying by doing this, but with no luck. 

`
roads = make_geocube(
        vector_data=roads,
        measurements=['train_id'],
        resolution=(-10,10),
        rasterize_function=partial(rasterize_image,all_touched=True,dtype=np.uint16)
    )
`
I can see a function called "_minimize_dtype()" in geocube.rasterize.py, which probably changes the data type to float32.

**Comment by snowman2** [comment, marked as answer]
Step 1: Make sure input data values are `uint16`
Step 2: Make sure the nodata value is compatible with `uint16` as the default is `numpy.nan`
https://corteva.github.io/geocube/stable/geocube.html#make-geocube
```python
roads["train_id"] = roads.train_id.astype(np.uint16)
roads = make_geocube(
    vector_data=roads,
    measurements=['train_id'],
    resolution=(-10,10),
    rasterize_function=partial(rasterize_image,all_touched=True),
    fill=0,
)
```
